packages/ui/ForwardRef/ForwardRef.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Provides a HOC interface for `ref` forwarding.
```

Comment:
Is the intent of this component to provide a more convenient way to simulate the upcoming React forwardRef feature that Enzyme doesn't support yet?

Reply:
To be clear, it's not _simulating_ `React.forwardRef`; it's using it from within the HOC.

Reply:
I like it. And just for verbose clarification, this _will_ work with the version of Enzyme we're using?

Reply:
Sadly, no. Hence the gymnastics included here

https://github.com/enactjs/enact/pull/1626/files#diff-598514a33c478cdac5391092c58009feR1662

Reply:
There's no way to simulate `forwardRef`. By attaching `ref` to a component instance, you will always get a reference to that instance and not any underlying component. The best we've accomplished previously is `ApiDecorator` which still returns a `ref` to itself by proxies method and property access.

`forwardRef` adds the internal plumbing necessary to take a `ref` prop in and send another prop out.